community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/RWLock.java

```diff
@@ -146,42 +172,53 @@ synchronized void acquireReadLock( Object tx ) throws DeadlockDetectedException
     {
         TxLockElement tle = getOrCreateLockElement( tx );
 
+        LockRequest lockRequest = null;
+        boolean addLockRequest = true;
```

Comment:
what does addLockRequest represent and why is it set as it is further down depending on whether an interrupt was raised or not?